sysdump/sysdump.go

```diff
@@ -963,13 +984,7 @@ func (c *Collector) Run() error {
 			Description:     "Collecting the CNI configuration files from Cilium pods",
 			Quick:           true,
 			Task: func(ctx context.Context) error {
-				p, err := c.Client.ListPods(ctx, c.Options.CiliumNamespace, metav1.ListOptions{
-					LabelSelector: c.Options.CiliumLabelSelector,
-				})
-				if err != nil {
-					return fmt.Errorf("failed to list Cilium pods: %w", err)
-				}
-				if err := c.SubmitCniConflistSubtask(FilterPods(p, c.NodeList), ciliumAgentContainerName); err != nil {
+				if err := c.SubmitCniConflistSubtask(c.CiliumPods, ciliumAgentContainerName); err != nil {
```

Comment:
Should we raise an error if the filtered list of pods is empty?

Reply:
i think what we should do is to not even start these tasks if there are no cilium pods anyways.

that would be a bigger change than i'd like to make in this pull request. let me do that as a follow up PR 🚀🙏